src/kaleidoscope/plugin/LEDControl.cpp

```diff
@@ -16,56 +16,64 @@
 
 #include "Kaleidoscope-LEDControl.h"
 #include "Kaleidoscope-FocusSerial.h"
+#include <Kaleidoscope-EEPROM-Settings.h>
 #include "kaleidoscope_internal/LEDModeManager.h"
 
 using namespace kaleidoscope::internal;
 
 namespace kaleidoscope {
 namespace plugin {
 
-static constexpr uint8_t uninitialized_mode_id = 255;
 
-uint8_t LEDControl::mode_id = uninitialized_mode_id;
+LEDControl::settings_t LEDControl::settings = {
+  .mode_id = 0, // defaults
```

Comment:
Tagged struct initialization is a C99 feature that is not (yet) available in C++. Here it works only because we add compiler flags that make the compiler more tolerant.